tests/unit/Owncloud/OcisPhpSdk/ResourceInviteTest.php

```diff
@@ -0,0 +1,162 @@
+                                    'object_id' => 'uuid-of-einstein',
+                                ]
+                            ),
+                        ],
+                        'roles' => ['uuid-of-the-role'],
+                    ]
+                )
+            ],
+            // invite a user and a group
+            [
+                [$einstein, $smartPeopleGroup],
+                null,
+                new DriveItemInvite(
+                    [
+                        'recipients' => [
+                            new DriveRecipient(
+                                [
+                                    'object_id' => 'uuid-of-einstein',
+                                ]
+                            ),
+                            new DriveRecipient(
+                                [
+                                    'object_id' => 'uuid-of-smart-people-group',
+                                    'at_libre_graph_recipient_type' => 'group',
+                                ]
+                            ),
+                        ],
+                        'roles' => ['uuid-of-the-role'],
+                    ]
+                )
+            ],
+            // set expiry time
+            [
+                [$smartPeopleGroup],
+                new \DateTime('2022-12-31 01:02:03.456789'),
+                new DriveItemInvite(
+                    [
+                        'recipients' => [
+                            new DriveRecipient(
+                                [
+                                    'object_id' => 'uuid-of-smart-people-group',
+                                    'at_libre_graph_recipient_type' => 'group',
+                                ]
+                            ),
+                        ],
+                        'roles' => ['uuid-of-the-role'],
+                        'expiration_date_time' => '2022-12-31T01:02:03:456789Z'
+                    ]
+                )
+            ],
+            // set expiry time, with conversion to UTC/Z timezone
+            [
+                [$einstein],
+                new \DateTime('2021-01-01 17:45:43.123456', new \DateTimeZone('Asia/Kathmandu')),
+                new DriveItemInvite(
+                    [
+                        'recipients' => [
+                            new DriveRecipient(
+                                [
+                                    'object_id' => 'uuid-of-einstein',
+                                ]
+                            ),
+                        ],
+                        'roles' => ['uuid-of-the-role'],
+                        'expiration_date_time' => '2021-01-01T12:00:43:123456Z'
+                    ]
+                )
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider inviteDataProvider
+     * @param array<int, User|Group> $recipients
+     */
+    public function testInvite($recipients, ?\DateTime $expiration, DriveItemInvite $expectedInviteData): void
```

Comment:
This unit test is semi-useful, but at least it does some basic checks. As soon as the underlying ocis has the functionality implemented we should add integration tests